Parse/Internal/File/Controller/PFFileController.m

```diff
@@ -218,21 +219,38 @@ - (NSString *)_temporaryFileDownloadPathForFileState:(PFFileState *)fileState {
         return [BFTask taskWithError:error];
     }
 
-    PFRESTFileCommand *command = [PFRESTFileCommand uploadCommandForFileWithName:fileState.name sessionToken:sessionToken];
-    @weakify(self);
-    return [[self.dataSource.commandRunner runFileUploadCommandAsync:command
-                                                     withContentType:fileState.mimeType
-                                               contentSourceFilePath:sourceFilePath
-                                                             options:PFCommandRunningOptionRetryIfFailed
-                                                   cancellationToken:cancellationToken
-                                                       progressBlock:progressBlock] continueWithSuccessBlock:^id(BFTask<PFCommandResult *> *task) {
-        @strongify(self);
-        PFCommandResult *result = task.result;
-        PFFileState *fileState = [[PFFileState alloc] initWithName:result.result[@"name"]
-                                                         urlString:result.result[@"url"]
-                                                          mimeType:nil];
-        return [[self _cacheFileAsyncWithState:fileState atPath:sourceFilePath] continueWithSuccessResult:fileState];
-    }];
+                         
+    if (uploadController) {
```

Comment:
why not have a default controller that proceeded the `PFRESTFileCommand`  this way the upload controller could be:

```
uploadController = uploadController ?: [PFRestFileUploadController default]
```